tests/test_dle.py

```diff
@@ -92,35 +92,32 @@ def test_cache_blocking_structure(blockinner, exp_calls, exp_iters):
                                              'par-collapse-ncores': 1}))
     calls = FindNodes(Call).visit(op)
     assert len(calls) == exp_calls
-    trees = retrieve_iteration_tree(op._func_table['bf0'].root)
+    trees = [i for i in retrieve_iteration_tree(op)]
     assert len(trees) == 1
     tree = trees[0]
     assert len(tree) == exp_iters
     if blockinner:
-        assert all(tree[i].dim.is_Incr for i in range(exp_iters))
+        assert all(tree[i].dim.is_Incr for i in range(1, exp_iters))
     else:
-        assert all(tree[i].dim.is_Incr for i in range(exp_iters-1))
+        assert all(tree[i].dim.is_Incr for i in range(1, exp_iters-1))
         assert not tree[-1].dim.is_Incr
 
     # Check presence of openmp pragmas at the right place
     _, op = _new_operator2((10, 31, 45), time_order=2,
                            opt=('blocking', {'openmp': True,
                                              'blockinner': blockinner,
                                              'par-collapse-ncores': 1}))
-    trees = retrieve_iteration_tree(op._func_table['bf0'].root)
+    trees = retrieve_iteration_tree(op)
     assert len(trees) == 1
     tree = trees[0]
-    assert len(tree.root.pragmas) == 1
-    assert 'omp for' in tree.root.pragmas[0].value
+    assert len(tree.root.pragmas) == 0
+    assert 'omp for' in trees[0][1].pragmas[0].value
```

Comment:
... as in, could we have more than 1 pragmas here?

Reply:
right, you mean to catch both 
```
START_TIMER(section0)
#pragma omp parallel num_threads(nthreads)
{
  #pragma omp for collapse(2) schedule(static,1)
  for (int x0_blk0 = x_m; x0_blk0 <= x_M; x0_blk0 += x0_blk0_size)
  {

```

Reply:
I think we're good now